core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -199,6 +210,25 @@ protected void doStop() throws Exception {
     ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
     context.setContextPath("/");
     context.addServlet(servletHolder, "/*");
+    String authMethod = getConfiguration().getString(RestConfig.AUTHENTICATION_METHOD);
+    if (!"NONE".equals(authMethod)) {
+      String realm = getConfiguration().getString(RestConfig.AUTHENTICATION_REALM);
+      List<String> roles = getConfiguration().getList(RestConfig.AUTHENTICATION_ROLES);
+      final ConstraintSecurityHandler securityHandler = new ConstraintSecurityHandler();
+      Constraint constraint = new Constraint();
+      constraint.setAuthenticate(true);
+      constraint.setRoles(roles.toArray(new String[roles.size()]));
+      ConstraintMapping constraintMapping = new ConstraintMapping();
+      constraintMapping.setConstraint(constraint);
+      constraintMapping.setMethod("*");
+      constraintMapping.setPathSpec("/*");
+      securityHandler.addConstraintMapping(constraintMapping);
```

Comment:
Yeah, I just want to make sure a) we don't get ourselves into an annoying compatibility mess and b) if we're going to express this via configs, that we *can* express this via configs. e.g. if we needed to annotate each resource handler and use introduce some additional info in the constraint somehow (e.g. "admin" vs "producer" vs "consumer" type resources in the REST proxy).